Add deprecations of 4.6 (#675)
kokkos/kokkos-core-wiki: docs/source/deprecations.rst

Deprecated in Kokkos 4.x
===========================

Deprecated in Kokkos 4.6
---------------------------

* ``StaticCrsGraph`` moved to Kokkos Kernels

>    * replacement: none

Reviewer: I'd list the corresponding name in `KokkosKernels`.
```suggestion
   * ``KokkosSparse::StaticCrsGraph``
```

>    * aligns better with functionality provoided by ``KokkosKernels``.

Reviewer: ```suggestion
   * aligns better with functionality provided by ``KokkosKernels``.
```

* ``native_simd`` type and ``simd_abi`` interface
   * replacement: none
   * alignment with the c++ standard

* Makefile support
   * replacement: CMake
   * transition to a modern build system

* Direct access to ``d_view`` and ``h_view`` members in ``DualView``
   * replacement: ``view_host()`` and ``view_device()``

>    * prevent assignment of a view

Reviewer: ```suggestion
   * enforcing invariants in ``DualView``, e.g., consistency between the ``View``s referenced
```

>    * enforcing invariants in ``DualView``, e.g., consistency between the ``View``s referenced

Reviewer 2: I think it is the "s" after `View` that's causing issues

>    * enforcing invariants in ``DualView``, e.g., consistency between the ``View``s referenced

Author: ```suggestion
   * enforcing invariants in ``DualView``, e.g., consistency between the two ``View`` instances referenced
```
